test/src/main/java/org/apache/accumulo/test/functional/MasterBulkFailureIT.java

```diff
@@ -0,0 +1,136 @@
+import org.apache.accumulo.core.conf.AccumuloConfiguration;
+import org.apache.accumulo.core.conf.Property;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Range;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.file.FileOperations;
+import org.apache.accumulo.core.file.FileSKVWriter;
+import org.apache.accumulo.core.file.rfile.RFile;
+import org.apache.accumulo.core.security.Authorizations;
+import org.apache.accumulo.core.util.CachedConfiguration;
+import org.apache.accumulo.harness.AccumuloClusterHarness;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.LocatedFileStatus;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.RemoteIterator;
+import org.apache.hadoop.io.Text;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class MasterBulkFailureIT extends AccumuloClusterHarness {
+  @Override
+  protected int defaultTimeoutSeconds() {
+    return 60;
+  }
+
+  @Test
+  public void test() throws TableExistsException, AccumuloSecurityException, AccumuloException,
+      TableNotFoundException, IOException {
+    final Connector c = getConnector();
+    final String tableName = getUniqueNames(1)[0];
+    c.tableOperations().create(tableName);
+    c.tableOperations().setProperty(tableName, Property.TABLE_MAJC_RATIO.getKey(), "1");
+    c.instanceOperations().setProperty(Property.TSERV_BULK_MAX_TABLET_OVERLAP.getKey(), "2");
+    // splits
+    SortedSet<Text> splits = new TreeSet<>();
+    for (int i = 1; i < 50; i++) {
+      splits.add(new Text(Integer.toString(i)));
+    }
+    c.tableOperations().addSplits(tableName, splits);
+
+    // create a bulk load file
+    final FileSystem fs = getCluster().getFileSystem();
+    final Path basePath = getCluster().getTemporaryPath();
+    CachedConfiguration.setInstance(fs.getConf());
+
+    final Path base = new Path(basePath, "testBulkLoad" + tableName);
+    fs.delete(base, true);
+    fs.mkdirs(base);
+
+    Path bulkFailures = new Path(base, "failures");
+    Path files = new Path(base, "files");
+    fs.mkdirs(bulkFailures);
+    fs.mkdirs(files);
+    FileSKVWriter writer = FileOperations.getInstance().newWriterBuilder()
+        .forFile(files.toString() + "/bulk." + RFile.EXTENSION, fs, fs.getConf())
+        .withTableConfiguration(AccumuloConfiguration.getDefaultConfiguration()).build();
+    writer.startDefaultLocalityGroup();
+    for (int j = 0; j < 5; j++) {
+      writer.append(new Key(Integer.toString(j)), new Value(new byte[0]));
+    }
+    writer.close();
+
+    Path file = new Path(files, "bulk." + RFile.EXTENSION);
+
+    c.tableOperations().importDirectory(tableName, files.toString(), bulkFailures.toString(),
+        false);
+
+    RemoteIterator<LocatedFileStatus> status = fs.listFiles(bulkFailures, false);
+    Assert.assertTrue(status.hasNext());
+    Path newPath = status.next().getPath();
+    Assert.assertFalse(status.hasNext());
```

Comment:
The bulkImport may not have run yet so you need to verify that the files were moved first before checking the failure directory. The same with the second try below.